src/components/shared/layout/layout.jsx

```diff
@@ -1,13 +1,21 @@
 import Nav from '../nav-bar/nav-bar';
 import Footer from '../footer/foot';
 import Newsletter from '../newsletter/newsletter';
+import { useRouter } from 'next/router';
 
 export default function Layout({ children }) {
+  const { route } = useRouter();
+
+  const mapToRouteName = {
```

Comment:
This variable name doesn't make sense here. You are not mapping anything to a route name, you are technically mapping a route to a boolean. 
To use this mapping naming convention correctly, you want to mention what the input value is, and what the return value is. In this case it would be `mapRouteToBoolean`, where `Route` is the input, and `Boolean` is the return value. 

Although a map works here, it also doesn't feel necessary. Consider a ternary:
```
const showNewsletter = route !== '/mail' && route !== '/contact'
```
I didnt test that line ^, you may have to use the `||` ternary instead of `&&`.